src/pages/api/auth/signup.ts

```diff
@@ -0,0 +1,34 @@
+import { NextApiHandler } from 'next';
+import { auth, adapter } from '@/firebase/client';
+import { createUserWithEmailAndPassword } from 'firebase/auth';
+import { AdapterUser } from 'next-auth/adapters';
+import { FirebaseError } from 'firebase/app';
+
+const handler: NextApiHandler = async (req, res) => {
+  if (req.method !== 'POST') {
+    return res.status(405);
+  }
+  const { firstname, lastname, password, email } = JSON.parse(req.body);
+
+  if (firstname && lastname && password && email) {
+    // Registration
+    try {
+      const data = await createUserWithEmailAndPassword(auth, email, password);
+      // if everything is good, create and return user
+      await adapter.createUser?.({
+        id: data.user.uid,
+        name: `${firstname} ${lastname}`,
+        email,
+        emailVerified: data.user.emailVerified ? new Date() : null,
+      } as AdapterUser);
+    } catch (err) {
+      const error = err as FirebaseError;
+      return res.status(500).json({ message: error.code });
+    }
+    return res.status(200).json({ text: 'OK' });
```

Comment:
I think you should return same objects like at code 500.

Since we returning the 200 code, frontend knows that request was fulfilled correctly. Let's rename it to { message }